bundle/edu.gemini.p2checker/src/main/java/edu/gemini/p2checker/rules/general/GeneralRule.java

```diff
@@ -326,6 +324,32 @@ public IP2Problems check(final ObservationElements elements)  {
         }
     };
 
+    /**
+     * Rule for asterism type: the asterism type must match one available for the instrument.
+     */
+    private static IRule ASTERISM_TYPE_RULE = elems -> {
+        final Option<TargetObsComp> tocOpt = elems.getTargetObsComp();
+        if (tocOpt.isEmpty())
+            return null;
+
+        final SPInstObsComp inst = elems.getInstrument();
+        if (inst == null)
+            return null;
+
+        final P2Problems problems = new P2Problems();
+        tocOpt.foreach(toc -> {
+            final SPComponentType instType = inst.getType();
+            final AsterismType    at       = toc.getAsterism().asterismType();
+            final ImList<AsterismType> supportedAsterismTypes = AsterismType.supportedTypesForInstrument(instType);
+            if (!supportedAsterismTypes.contains(at))
+                problems.addError(PREFIX + "ASTERISM_TYPE_RULE",
+                        instType.readableStr + " observation is configured to use an unsupported asterism of type " + at.name + ".",
+                        elems.getTargetObsComponentNode().getValue());
```

Comment:
It seems like we should indicate how to fix this problem if we can?  It might not be easy to fix of course.  There's no way in the target component to change the asterism type.  Even if you delete the target component and add another you likely won't have fixed it, unless we hack around in the OT to check and possibly change the asterism type when you add a target component.

In the merge conflict notes that we add, there are links to fix problems automatically.  I wonder if we could do something like that.  We can come back to this problem later I suppose though, since it really isn't very important for September.

Reply:
Could you add a task for this?

Reply:
Sure thing! I'll add a GHOST subtask.